samples/manageServices.py

```diff
@@ -0,0 +1,262 @@
+    parser.add_argument("-lc", "--listCommands", help="list available commands", action="store_true")
+    parser.add_argument("-sn", "--snapshotName", help="name if snapshot to create after changes")
+    parser.add_argument("-vf", "--vmFile", help="text file with vms to apply changes")
+    parser.add_argument("-w", "--wait", help="override the time (in seconds) to wait after running commands/script")
+    parser.add_argument("-cf", "--credsFile", help="credentials file for logging into the vm")
+    parser.add_argument("-af", "--actionFile", help="json file containing commands/actions")
+    parser.add_argument("-un", "--username", help="vm username")
+    parser.add_argument("-pw", "--password", help="vm password")
+    parser.add_argument("-t", "--threads", help="concurrent vms to alter")
+    parser.add_argument("hypervisorConfig", help="json hypervisor config")
+    args = parser.parse_args()
+
+    """
+    PREP COMMAND FILE
+    """
+    if args.actionFile != None:
+        commandFile = args.actionFile
+    else:
+        commandFile = "./action_scripts/commands.json"
+    commandDictionary = sampleLib.loadJsonFile(commandFile)
+    if commandDictionary == None:
+        print("FAILED TO LOAD COMMANDS")
+        return 0
+    """
+    IF THEY ASK TO LIST COMMANDS, DO IT AND EXIT
+    """
+    if args.listCommands:
+        return listCommands(commandDictionary)
+    
+    """
+    ENSURE THE COMMAND EXISTS
+    """
+    if args.action not in commandDictionary:
+        print("SELECTED ACTION NOT LISTED IN COMMAND FILE")
+        listCommands(commandDictionary)
+        return 0
+    
+    """
+    CHECK THAT WE HAVE REQUIRED CREDS SOMEHOW
+    """
+    if (args.username == None or args.password == None) and args.credsFile == None:
+        print("VM CREDENTIALS REQUIRED FOR THIS OPERATION")
+        exit(0)
+
+    credsDictionary = None
+    if args.credsFile != None:
+        credsDictionary = sampleLib.loadJsonFile(args.credsFile)
+        if credsDictionary == None:
+            print("FAILED TO LOAD CREDS FILE")
+            exit(0)
+    
+    replaceUsername(commandDictionary[args.action], args.username)
+    replacePassword(commandDictionary[args.action], args.password)
+    """
+    UPDATE WAIT, IF REQUIRED
+    """
+    if args.wait != None:
+        commandDictionary[args.action]['WAIT_SECONDS'] = int(args.wait)
+        
+    """
+    CREATE SERVER
+    """
+    logFile = './logs/service_management.log'
+    try:
+        os.remove(logFile)
+    except:
+        pass
+    vmServer = vm_automation.esxiServer.createFromFile(args.hypervisorConfig, logFile)
+    if vmServer == None:
+        print("VM SERVER CREATION FAILED")
+        return 0
+    
+    """
+    GENERATE LIST OF VMS THAT NEED TO CHANGE
+    """
+    vmsToChange = sampleLib.makeVmList(vmServer, args.keyword, args.vmFile)
+    
+    """
+    ARE WE USING THREADS?
+    """
+    if args.threads != None:
```

Comment:
instead of having duplication of the serial case the overhead of a pool with only one executor seems reasonable for the code it removes.

```
num_threads = 1
if args.threads != None:
    num_threads = args.threads
```

Reply:
I'm OK with that.